aips/aip-113.md

```diff
@@ -45,67 +50,66 @@ Derivable AA introduces the following features:
 
 ## Specification and Implementation Details
 
-### Domain Dispatchable Authenticator
+### Derivable Dispatchable Authenticator
 
-The `DomainDispatchableAuthenticator` is a resource that manages authentication functions for a specific domain. This resource at `@0x1` holds a whitelist of authentication functions that can be used by accounts within the domain.
+The `DerivableDispatchableAuthenticator` is a resource that manages authentication functions for a specific domain. This resource at `@0x1` holds a whitelist of authentication functions that can be used by accounts within the domain.
 
 ```rust
-#[resource_group_member(group = aptos_framework::object::ObjectGroup)]
-/// Domain-scoped authenticator that defines how to authenticate accounts in the specified domain.
-/// An integral part of Domain Account Abstraction.
-enum DomainDispatchableAuthenticator has key {
-    V1 { auth_functions: BigOrderedMap<FunctionInfo, bool> }
-}
+    /// The dispatchable derivable-scoped authenticator, that defines how to authenticate
+    enum DerivableDispatchableAuthenticator has key {
+        V1 { auth_functions: BigOrderedMap<FunctionInfo, DerivableRegisterValue> }
+    }
 ```
 
 ### Account Address Derivation
-The sender account address of Derivable AA is derived based on the `account_identity` and the `function_info` for the domain.
+The sender account address of Derivable AA is derived based on the `abstract_public_key` and the `function_info`.
 ```rust
-#[view]
-/// Derives an account address for Derivable AA based on the module address, function name, and account identity.
-public fun domain_aa_account_address(
-    module_address: address,
-    module_name: String,
-    function_name: String,
-    account_identity: vector<u8>
-): address {
-    let function_info = function_info::new_function_info_from_address(module_address, module_name, function_name);
-    let bytes = bcs::to_bytes(&function_info);
-    bytes.append(bcs::to_bytes(account_identity));
-    bytes.push_back(DOMAIN_ABSTRACTION_DERIVED_SCHEME);
-    from_bcs::to_address(hash::sha3_256(bytes))
-}
+    /// Return the account address corresponding to the given `abstract_public_key`,
+    /// for the derivable account abstraction defined by the given function.
+    public fun derive_account_address(derivable_func_info: FunctionInfo, abstract_public_key: &vector<u8>): address {
```

Comment:
Wouldn't "authentication key" be the correct term here as opposed to "account address"?
In most cases it's the same as the account's address, but if an account is rotated to a new derivable abstract public key this wouldn't work

Reply:
this is account address - which cannot be changed. so you permanently need to use fixed abstract_public_key.

For rotation:
- We currently don't support "disconnecting" account from original public key, though we can add that.
- You can account::rotate_authentication_key_call, and add another native key to it.
- implementation could support other rotations internally, SIWS implementation doesn't support that